### DISCUSSION TITLE
using path param of type union breaks swaggerui

### DISCUSSION THREAD

**Discussion by dmig-alarstudios** [fastapi/fastapi#7737] [Questions] [answered]
### Describe the bug

Noticed this bug after 0.46 -> 0.47.1 update. Endpoint using a path parameter of type `Union[int, str]` causes Swagger UI 'try it out' to break emitting error in system.js:464 and showing spinner indefinitely:
```
Error: Required parameter param is not provided
    at index.js:3893
    at Array.forEach (<anonymous>)
    at Object.jn [as buildRequest] (index.js:3863)
    at actions.js:415
    at utils.js:137
    at bindActionCreators.js:3
    at wrap-actions.js:33
    at Object.r (system.js:174)
    at Object.executeRequest (system.js:461)
    at actions.js:458
```

### To Reproduce

Steps to reproduce the behavior with a minimum self-contained file.

Replace each part with your own scenario:

1. Create a file with:

```Python
from typing import Union
from fastapi import FastAPI

app = FastAPI()

@app.get('/{param}')
def test(param: Union[int, str]):
    return param
```

3. Open the browser and call the endpoint `/docs`.
4. Press 'Try it out' of your endpoint.
5. Press 'Execute' without filling in the parameter.

### Expected behavior

Notification about empty required parameter.

### Environment

- OS: Ubuntu 19.10, google chrome 79.0.3945.130
- FastAPI: 0.47.1
- Python: 3.8.1 installed with `pyenv`

**Comment by prostomarkeloff** [comment]
Can't reproduce using `Firefox 72.0.1`

**Comment by dmig-alarstudios** [comment]
Probably Chrome related.

![Screenshot from 2020-01-27 09-43-14](https://user-images.githubusercontent.com/50368846/73160559-9df72980-40e9-11ea-9d27-b65ced85308c.png)
![Screenshot from 2020-01-27 09-43-23](https://user-images.githubusercontent.com/50368846/73160565-a2234700-40e9-11ea-868b-bcc4cdd5bf70.png)

**Comment by tiangolo** [comment, marked as answer]
Oh! That's fun. I was able to reproduce it. That's strange. But it's actually not an issue with FastAPI but with Swagger UI.

I added the issue here: https://github.com/swagger-api/swagger-ui/issues/5831

I'll close this issue as it is not really on the FastAPI side. We can track it there.
